### DISCUSSION TITLE
how to read a text file in my github repository

### DISCUSSION THREAD

**Discussion by mrslamf** [community/community#46797] [New to GitHub]
I have a Java application that I want to read a text file from my github repository.   

import static JavaManifestFormatter.JavaManifestFormatter.versionFilePath;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.logging.Level;
import java.util.logging.Logger;

public class checkVersion {
    
    public static void checkVersion(){
        String[] inputLine = new String[500];
        int counter = 0;

        try {
            URL targetUrl = new URL (versionFilePath); 
              
            BufferedReader read = new BufferedReader(new InputStreamReader(targetUrl.openStream()));

            while ((inputLine[counter] = read.readLine()) != null){
                System.out.println("# " + counter + "  " + inputLine[counter]);
                counter++;
            }
            
            read.close();
            
        } catch (MalformedURLException ex) {
            Logger.getLogger(checkVersion.class.getName()).log(Level.SEVERE, null, ex);
            utilities.writeToResultsWindow(1, "MalformedURLException " + ex);            
        }
        catch (IOException ex) {
            Logger.getLogger(checkVersion.class.getName()).log(Level.SEVERE, null, ex);
            utilities.writeToResultsWindow(1, "IOException " + ex);
        }     
}
}

The code runs without error but  reads in a bunch of html code instead of the file's contents.  
# 0  
# 1  
# 2  
# 3  
# 4  
# 5  
# 6  
# 7  <!DOCTYPE html>
# 8  <html lang="en" data-color-mode="auto" data-light-theme="light" data-dark-theme="dark" data-a11y-animated-images="system">
# 9    <head>
# 10      <meta charset="utf-8">
# 11    <link rel="dns-prefetch" href="https://github.githubassets.com">
# 12    <link rel="dns-prefetch" href="https://avatars.githubusercontent.com">
# 13    <link rel="dns-prefetch" href="https://github-cloud.s3.amazonaws.com">
# 14    <link rel="dns-prefetch" href="https://user-images.githubusercontent.com/">
# 15    <link rel="preconnect" href="https://github.githubassets.com" crossorigin>
# 16    <link rel="preconnect" href="https://avatars.githubusercontent.com">

 and on and on for > 500 lines

How do I read the contents of the file (it is a single numeric value)?

Thanks

Tim

**Comment by DevBadger10** [comment]
I don't have any knowledge of java, but are you using the url you get if you enter the .txt file in GitHub then press ... and then Veiw Raw? This *might* be your problem.

**Reply by mrslamf** [reply]
I have tried the link I get when I open the file in GitHub and press the "..." and then copy file path and I also tried the permalink

**Comment by Juanescacha** [comment]
I don't know about java but you can achieve this and more with web scrapping, give it a look at this website if it helps you
https://www.zenrows.com/blog/web-scraping-java#:~:text=You%20can%20scrape%20a%20web,library%20with%20Maven%20or%20Gradle.
